### ISSUE TITLE
alias alphabetization not honoring `:as`

### ISSUE THREAD

**Issue by asummers** [rrrene/credo#616]
### Environment

* Credo version (`mix credo -v`): 1.0.0
* Erlang/Elixir version (`elixir -v`):
```
Erlang/OTP 22 [DEVELOPMENT] [erts-10.0.1] [source-70b46d747d] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe]

Elixir 1.7.4 (compiled with Erlang/OTP 22)
```

* Operating system: OSX

### What were you trying to do?

Consider the following program:

```elixir
defmodule ABC.CDE do
  @moduledoc false
  ##
end

defmodule ABC.CDE.EFG do
  @moduledoc false
  ##
end

defmodule Example do
  @moduledoc false
  alias ABC.CDE, as: AA
  alias AA.EFG
  ##
end
```

### Expected outcome
When running `mix credo --strict` there will be no alias alphabetization warning.

### Actual outcome

```
❯ mix credo --strict --format=oneline
[R] ↘ lib/example.ex:14:9 The alias `AA.EFG` is not alphabetically ordered among its group.
```

Because `AA` follows `ABC` in alias order but precedes `ABC` in alphabetical order, credo suggests `AA.EFG` to precede `ABC.CDE, as: AA`. This is obviously impossible, as moving that above no longer creates a valid `alias` without unfolding the `:as` explicitly. 

Now, it's questionable on whether this type of `:as alias` should be done, but the error produced is incorrect, to me, considering that it is allowed by the language. As a writer, I would generally unfold the second alias to `alias ABC.CDE.EFG`, but that should be a separate rule, if that is something Credo is suggesting, though I do not believe that is explicit in this case.

**Comment by rrrene** [comment]
@asummers I did not understand the issue at first, but you are right: If the language supports this, we should honor/support it. 

I would accept a PR for this. :+1:

**Comment by asummers** [comment]
That code is a little more involved than I was expecting. Wrapping my head around it, but it looks like it should eventually be straightforward to add. It seems like some things can be a bit simpler as well, given this new constraint, so I may try to kill two birds with one stone. Will update when I have something.

**Comment by asummers** [comment]
Okay so in my version locally, I have a test failure I don't think is actually correct. In `Credo.Check.Readability.AliasOrderTest` L10, why would `alias Credo.CLI.ExitStatus` not be alphabetically before `alias Credo.Check` and friends? 

```elixir
iex(1)> CLI < Check
true
iex(2)> "CLI" < "Check"
true
```

I believe this to be a bug in the current test 

Additionally, I have it producing two failures in L119 with 

```
      alias Credo.CLI.Sorter
      alias Credo.CLI.Command
      alias Credo.CLI.Filename
```

because both `Command` and `Filename` are out of order. Is this fine? Will try to restore to previous behavior if possible, of course, but wondering how much I need to bother, if at all.

**Comment by rrrene** [comment]
@asummers 

* yes, L10 should not raise any issues
* as of now, L119 should raise a *single* issue (since `Command` and `Filename` are out of order, as you mentioned), so the test suite fails if the check raises two issues - but one might argue that raising two issues is philosophically "more correct" here 😄

**Comment by asummers** [comment]
I guess I'm completely unclear why L10 should not have any issues. By Elixir ABC rules, `Credo.CLI.ExitStatus` comes before `Credo.Check.*`. What is the metric by which that should be alphabetically before?

**Comment by rrrene** [comment]
@asummers Just so we're clear, we are talking about this example:

```elixir
defmodule Test do
  alias App.Module1
  alias App.Module2
  alias Credo.CLI.Command
  alias Credo.CLI.Filename
  alias Credo.CLI.Sorter

  alias Credo.Check
  alias Credo.Check.CodeHelper
  alias Credo.Check.Params
  alias Credo.CLI.ExitStatus
  alias Credo.Issue
end
```

Blocks should be alphabetically ordered (case-insensitive), which they are in this example. `Credo.Check` comes before `Credo.CLI`, because `H` comes before `L` (using uppercase letters for easier readability).

The check is case-insensitive because I think that humans do not associate "sort it alphabetically" with "sort it like an ASCII table".

Do we really want the following?

```elixir
defmodule Test do
  alias Credo.CCTV
  alias Credo.Check
  alias Credo.CcEmail
end
```

**Comment by asummers** [comment]
Yes, that is the one I am referring to. In  that example it would be:

```
  alias Credo.CCTV
  alias Credo.CcEmail
  alias Credo.Check
```

since `Cc < Ch`. That seems correct to me. I can see the argument about casing somewhat, but given that module AND string sorts on that yield that order without normalizing case, any other behavior is divergent from that. I would expect as a user that if I stringified  the aliases and threw them into a list, and called `Enum.sort` on it, that the aliases should be in that order.

**Comment by NobbZ** [comment]
I'd like to see email before TV in that example, as E is before T.

**Comment by rrrene** [comment]
Hi, thx again for this. Unfortunately, I am closing it due to "it being obsolete" IMHO. 

If you disagree, please re-open and elaborate! :+1:
